pkg/sessionctx/variable/sysvar.go

```diff
@@ -957,8 +957,11 @@ var defaultSysVars = []*SysVar{
 			}
 			AutoAnalyzePartitionBatchSize.Store(val)
 			return nil
-		}},
-
+		}, Validation: func(vars *SessionVars, normalizedValue string, originalValue string, scope ScopeFlag) (string, error) {
+			vars.StmtCtx.AppendWarning(ErrWarnDeprecatedSyntaxNoReplacement.FastGenByArgs(TiDBAutoAnalyzePartitionBatchSize))
```

Comment:
Please add comments to mark TiDBAutoAnalyzePartitionBatchSize as deprecated.

Reply:
I have added comments.